sansa-owl-spark/src/main/scala/net/sansa_stack/owl/spark/stats/OWLStats.scala

```diff
@@ -0,0 +1,821 @@
+
+  /**
+    * 28.Maximum value per property {int,float,double} criterion
+    *
+    * @param axioms RDD of OWLAxioms
+    * @return entities with their maximum values
+    */
+  def getMaxPerNumericDatatypeProperty(axioms: RDD[OWLAxiom]): RDD[(OWLDataPropertyExpression, Int)] = {
+
+    val dataPropAssrNumbers = owlAxioms.extractAxioms(axioms, AxiomType.DATA_PROPERTY_ASSERTION)
+                                .asInstanceOf[RDD[OWLDataPropertyAssertionAxiom]]
+                                .filter(a =>
+                                  a.getObject.getDatatype.isInteger ||
+                                    a.getObject.getDatatype.isDouble ||
+                                    a.getObject.getDatatype.isFloat)
+                                .map(a => (a.getProperty, a.getObject.getLiteral.toInt))
+
+    val negDataPropAssrNumbers = owlAxioms.extractAxioms(axioms, AxiomType.NEGATIVE_DATA_PROPERTY_ASSERTION)
+                                   .asInstanceOf[RDD[OWLNegativeDataPropertyAssertionAxiom]]
+                                   .filter(a =>
+                                      a.getObject.getDatatype.isInteger ||
+                                        a.getObject.getDatatype.isDouble ||
+                                        a.getObject.getDatatype.isFloat)
+                                   .map(a => (a.getProperty, a.getObject.getLiteral.toInt))
+
+    val max = dataPropAssrNumbers.union(negDataPropAssrNumbers)
+                          .reduceByKey(_ max _)
+
+    max
+  }
+
+  /**
+    * Criterion 29. Average value per numeric property {int,float,double}
+    *
+    * @param axioms RDD of OWLAxioms
+    * @return properties with their average values
+    */
+  def getAvgPerNumericDatatypeProperty(axioms: RDD[OWLAxiom]): RDD[(OWLDataPropertyExpression, Double)] = {
+    val dataPropAssrLiterals = owlAxioms.extractAxioms(axioms, AxiomType.DATA_PROPERTY_ASSERTION)
+                                     .asInstanceOf[RDD[OWLDataPropertyAssertionAxiom]]
+                                     .filter(a =>
+                                       a.getObject.getDatatype.isInteger ||
+                                         a.getObject.getDatatype.isDouble ||
+                                         a.getObject.getDatatype.isFloat)
+                                     .map(a => (a.getProperty, a.getObject))
+
+    val negDataPropAssrLiterals = owlAxioms.extractAxioms(axioms, AxiomType.NEGATIVE_DATA_PROPERTY_ASSERTION)
+                                   .asInstanceOf[RDD[OWLNegativeDataPropertyAssertionAxiom]]
+                                   .filter(a =>
+                                     a.getObject.getDatatype.isInteger ||
+                                       a.getObject.getDatatype.isDouble ||
+                                       a.getObject.getDatatype.isFloat)
+                                   .map(a => (a.getProperty, a.getObject))
+
+    val average = dataPropAssrLiterals.union(negDataPropAssrLiterals)
```

Comment:
Why not just use `.mean` here?

Reply:
I found no direct `.mean` method. I change the code to avoid `aggregateByKey`. Will push it soon